ApiReview/API.Polly.Core/NoDocs/Polly/ResiliencePropertyKey.cs

```diff
@@ -0,0 +1,18 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Polly;
+
+public readonly struct ResiliencePropertyKey<TValue> : IEquatable<ResiliencePropertyKey<TValue>>
```

Comment:
what is the purpose of this type?
does it imply that in order to prevent the case where one populates a dictionary with an entry of one type but retrieves it as a different type one must use the same ResiliencePropertyKey instance?

Reply:
It's to encourage the reuse of keys as opposed to providing them in each access and to avoid using "magic strings".

> does it imply that in order to prevent the case where one populates a dictionary with an entry of one type but retrieves it as a different type one must use the same ResiliencePropertyKey instance?

Yes, the type of the result is tied to the key name, thus preventing the scenario above. It also simplifies the usage as consumer doesn't have to cast the value anymore.

We "stole" this API from here:
https://source.dot.net/#System.Net.Http/System/Net/Http/HttpRequestOptionsKey.cs,12

Reply:
Explained the usage and origin and we agreed on the API and it's usage.

Reply:
wait